erpnext/controllers/stock_controller.py

```diff
@@ -342,6 +342,18 @@ def validate_inspection(self):
 				if self.docstatus==1:
 					raise frappe.ValidationError
 
+	def update_blanket_order(self, cancel=False):
+		for item in self.items:
+			if item.blanket_order:
+				ordered_quantity, doc_name = frappe.db.get_value("Blanket Order Item", {"parent": item.blanket_order}, ["ordered_quantity", "name"])
+				if not cancel:
+					ordered_quantity = ordered_quantity + item.qty
```

Comment:
Write this type function using `select sum(qty)` query, this helps to correct if goes wrong. For example, you forgot to call this function on STOP and UNSTOP events.